### PR TITLE
Add the capability to assimilate the MADIS snow depth for global-workflow

### PR DESCRIPTION
**Pull request by jiaruidong2017**
This PR adds `bufr_snocvr.yaml` to covert GTS based `madis` snow depth data into IODA format and adds `snocvr_snow.yaml` to enable the global-workflow to assimilate the madis snow depth data from GTS. 

This PR also updates the configures `gdas_land_gts_only.yaml` and `gdas_land_prototype.yaml` to include the new `madis` snow depth data for global-workflow. 

This PR contributes to the https://github.com/NOAA-EMC/global-workflow/pull/1836 for assimilating the MADIS snow depth.

### CODE REVIEW

File: parm/land/obs/config/snocvr_snow.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+obs space:
+  name: snocvr_snow
+  distribution:
+    name: Halo
+    halo size: 250e3
+  obsdatain:
+    engine:
+      type: H5File
+      obsfile: $(DATA)/obs/$(OPREFIX)snocvr_snow.nc4
```
Comment (aerorahul) [inline parm/land/obs/config/snocvr_snow.yaml]:
```suggestion
      obsfile: {{ DATA }}/obs/{{ OPREFIX }}snocvr_snow.nc4
```

File: test/testinput/bufr_snocvr_snow.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
+# (C) Copyright 2021-2022 NOAA/NWS/NCEP/EMC
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
```
Comment (aerorahul) [inline test/testinput/bufr_snocvr_snow.yaml]: Just noting that the project is under an LGPL license and this is an Apache banner.

**Review by aerorahul — COMMENTED** [review]
Changes look good to me.
I have one suggestion -- we have been trying to move away from the `$( )` syntax in yaml and replace it with `{{ }}`.  It is still necessary to have the `${ }` syntax, especially when including nested yamls.  Would you be able and willing to replace `$( )` with `{{ }}` in this PR?

File: parm/land/obs/config/snocvr_snow.yaml (hunk 1 above)
Comment (jiaruidong2017) [reply parm/land/obs/config/snocvr_snow.yaml]: Thanks @aerorahul I made the changes from `$()` to `{{ }}` as suggested, and the test run worked well. The changes were committed to the repo.

File: test/testinput/bufr_snocvr_snow.yaml (hunk 2 above)
Comment (jiaruidong2017) [reply test/testinput/bufr_snocvr_snow.yaml]: @CoryMartin-NOAA Do you have any idea on this? Thanks.

Comment (CoryMartin-NOAA) [reply test/testinput/bufr_snocvr_snow.yaml]: Delete it all, we don't need a license in every file if the top level project has one.

File: parm/land/prep/prep_gts.yaml [hunk 3]
```diff
@@ -3,7 +3,11 @@ gtsbufr:
     - $(DATA)/obs
   copy:
     - [$(COM_OBS)/$(OPREFIX)adpsfc.tm00.bufr_d, $(DATA)/obs/]
+    - [$(COM_OBS)/$(OPREFIX)snocvr.tm00.bufr_d, $(DATA)/obs/]
```
Comment (aerorahul) [inline parm/land/prep/prep_gts.yaml]: can you also replace the `$( )` with `{{ }}` in this yaml?

File: test/testinput/bufr_snocvr_snow.yaml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+observations:
+  - obs space:
+      name: bufr
+
+      obsdatain: $(DATA)/obs/$(OPREFIX)snocvr.tm00.bufr_d
```
Comment (aerorahul) [inline test/testinput/bufr_snocvr_snow.yaml]:
```suggestion
      obsdatain: {{ DATA }}/obs/{{ OPREFIX }}snocvr.tm00.bufr_d
```

Comment (aerorahul) [reply test/testinput/bufr_snocvr_snow.yaml]: and elsewhere in this yaml

File: parm/land/prep/prep_gts.yaml (hunk 3 above)
Comment (jiaruidong2017) [reply parm/land/prep/prep_gts.yaml]: Made changes.

File: test/testinput/bufr_snocvr_snow.yaml (hunk 4 above)
Comment (jiaruidong2017) [reply test/testinput/bufr_snocvr_snow.yaml]: Made all the changes.

File: test/testinput/bufr_snocvr_snow.yaml (hunk 2 above)
Comment (jiaruidong2017) [reply test/testinput/bufr_snocvr_snow.yaml]: Thanks @CoryMartin-NOAA The comments were removed.

**Review by aerorahul — APPROVED** [review]
lgtm

**Comment by jiaruidong2017** [comment]
@aerorahul Thanks for quick reviews.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**